dig.go

```diff
@@ -579,8 +632,8 @@ func (c *Container) Invoke(function interface{}, opts ...InvokeOption) error {
 	}
 
 	if !c.isVerifiedAcyclic {
-		if err := c.verifyAcyclic(); err != nil {
-			return err
+		if ok, cycle := graph.IsAcyclic(c.gh); !ok {
+			return errf("cycle detected in dependency graph", c.cycleDetectedError(cycle))
```

Comment:
minor unintentional change in behavior here:
We won't mark the container as acyclic `if ok`, which we were doing before.
We should probably still mark it acyclic afterwards so that we don't check again for every invoke until it's dirty again.